http/src/main/resources/jupyter/gce-init.sh

```diff
@@ -0,0 +1,537 @@
+#!/usr/bin/env bash
+
+# This init script instantiates the tool (e.g. Jupyter) docker images on the Dataproc cluster master node.
+# Adapted from https://github.com/GoogleCloudPlatform/dataproc-initialization-actions/blob/master/datalab/datalab.sh
+
+set -e -x
+
+#####################################################################################################
+# Functions
+#####################################################################################################
+
+# Retry a command up to a specific number of times until it exits successfully,
+# with exponential back off. For example:
+#
+#   $ retry 5 echo "Hello"
+#     Hello
+#
+#   $ retry 5 false
+#     Retry 1/5 exited 1, retrying in 2 seconds...
+#     Retry 2/5 exited 1, retrying in 4 seconds...
+#     Retry 3/5 exited 1, retrying in 8 seconds...
+#     Retry 4/5 exited 1, retrying in 16 seconds...
+#     Retry 5/5 exited 1, no more retries left.
+function retry {
+  local retries=$1
+  shift
+
+  for ((i = 1; i <= $retries; i++)); do
+    # run with an 'or' so set -e doesn't abort the bash script on errors
+    exit=0
+    "$@" || exit=$?
+    if [ $exit -eq 0 ]; then
+      return 0
+    fi
+    wait=$((2 ** $i))
+    if [ $i -eq $retries ]; then
+      log "Retry $i/$retries exited $exit, no more retries left."
+      break
+    fi
+    log "Retry $i/$retries exited $exit, retrying in $wait seconds..."
+    sleep $wait
+  done
+  return 1
+}
+
+function log() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@"
+}
+
+function betterAptGet() {
```

Comment:
I don't believe this function is used by the init script, maybe we should remove it.

Reply:
👍